CIP-XXXX/README.md

```diff
@@ -0,0 +1,116 @@
+
+This operation appears in many cryptographic protocols. Its naive implementation requires $N$ scalar multiplications and $N$ group additions. However, the performance can be significantly improved by employing advanced algorithms, such as the [Pippenger Approach](https://hackmd.io/@tazAymRSQCGXTUKkbh1BAg/Sk27liTW9). Moreover, it can be further optimized for a particular group type (e.g., for elliptic curve groups [[BH22](https://eprint.iacr.org/2022/1400.pdf), [L23](https://dspacemainprd01.lib.uwaterloo.ca/server/api/core/bitstreams/3b15ca4c-9125-4e45-9378-c5474eec6a07/content)]).
+
+Multi-scalar multiplication appears in various cryptographic protocols, such as cryptographic signatures, zero-knowledge proofs, SNARK systems, and others. It is especially important in elliptic-curve-based SNARK proof systems, where large-scale MSMs can become a bottleneck in both proving and verification algorithms.
+
+The recent Chang upgrade in Cardano included [CIP-0381](https://cips.cardano.org/cip/CIP-0381), which introduced built-in support for operations over the BLS12-381 elliptic curve (the implementation uses [blst](https://github.com/supranational/blst/tree/master) library). It made it feasible to implement various SNARK systems on-chain in Cardano. However, MSM still remains a bottleneck for many use cases. Implementing MSM naively, even using built-in functions for BLS12-381, consumes a significant portion of the computational budget of a transaction. It hinders implementation of such SNARK systems as KZG-based PLONK or Groth16, which require computations of MSM.
+
+Our [benchmarks](https://github.com/input-output-hk/plutus-msm-bench) show that MSM of 10 $G1$ points over BLS12-381 curve consumes 7.74% of the computational budget of a transaction, while MSM of size more than 129 cannot fit into a single transaction at all. It impedes verification of complex circuits which might require much larger MSM.
+
+We also did preliminary [benchmarks](https://github.com/dkaidalov/bench-blst-msm/) to compare an [optimized blst](https://github.com/supranational/blst/blob/e99f7db0db413e2efefcfd077a4e335766f39c27/src/multi_scalar.c) implementation of MSM with naive implementation using just blst group operations. We used Rust bindings to do these benchmarks (the underlying bindings are the same as used in [cardano-base for bslt](https://github.com/IntersectMBO/cardano-base/blob/master/cardano-crypto-class/src/Cardano/Crypto/EllipticCurve/BLS12_381/Internal.hs#L353), which means we can expect similar behavior for the Haskell stack). The results for G1 group are the following (more results in the [repository](https://github.com/dkaidalov/bench-blst-msm/)):
+
+| MSM Size | MSM Optimized Average Time | MSM Naive Average Time | Ratio Naive/Optimized |
```

Comment:
Can we also get the spread on these numbers for random points? Note that the plutus benchmarks also take this spread into account for the adversarial case (as the costing is generally an overestimate of the real cost). 

I do not think it will be bad, just want to check it in advance so that we have no surprises once the plutus benchmarks come.